### ISSUE TITLE
custom elements not working inside <p> tags?

### ISSUE THREAD

**Issue by jlipps** [enhance-dev/enhance-starter-project#10]
I have a custom element `<ud-listen-list>`. It works as expected in this context:

```html
<div><ud-listen-list></ud-listen-list></div>
```

The inlined html includes both the surrounding div tag as well as the custom element and what's inside it:

```html
<!-- generated html -->
<div>
  <ud-listen-list>
    <!-- internals -->
  </ud-listen-list>
</div>
```

but if I change `<div>` to `<p>`, i get weird, incorrect html generated:

```html
<p>
  <ud-listen-list></ud-listen-list>
</p>
<!-- internals -->
```

i.e., the stuff inside the custom element gets inlined outside of the `<p>`. this is incorrect structure and also breaks styling rules.

**Comment by tbeseda** [comment]
### tldr: I encountered the same thing. I will try to recreate a minimal test case.

---

Ooh! I ran into something similar yesterday working on enhance.dev docs site.

I created a custom element `doc-video` that had contents like a `cite` tag and a vendored custom element `mux-player`
When I authored
```html
<p>
  <doc-video video-id="a-mux-playback-id"></doc-video>
</p>
```
The browser would render
```html
<p>
  <doc-video video-id="a-mux-playback-id"></doc-video>
</p>
<cite>my citation</cite>
<mux-player><!-- all the mux player internals --></mux-player>
```
Super weird part: when I looked at the raw HTML sent from the server I saw what I expected:
```html
<p>
  <doc-video video-id="a-mux-playback-id">
    <cite>my citation</cite>
    <mux-player><!-- all the mux player internals --></mux-player>
  </doc-video>
</p>
```

I chalked it up to being an issue with the Markdown renderer that sat between my authored content and Enhance.
I started working in a "clean room" plain HTML file without the parent `<p>` tag. There it worked as expected.

When I finished working on the element I tried it again in the main docs site and it worked fine, so I figured I had a typo somewhere.

No idea what changed to make it work. I will create a simple Enhance project and try to recreate this issue so we can investigate.

**Comment by tbeseda** [comment]
We solved this in another chat channel.
Turns out custom elements aren't allowed inside a `<p>` tag. Doh!
Enhance (powered by parse5) sticks to the spec and will kick custom elements out of nodes where they are not allowed.
We'll update docs to mention this as it's probably a common problem as more and more sites are built with custom HTML elements.
